search-commons/src/main/java/no/unit/nva/search2/ResourceQuery.java

```diff
@@ -40,24 +45,68 @@ private ResourceQuery() {
 
     @SuppressWarnings("PMD.NullAssignment")
     @NotNull
-    private PagedSearchResourceDto toPagedSearchResponseDto(@NotNull OpenSearchSwsResponse response) {
+    public PagedSearchResourceDto toPagedSearchResponseDto(@NotNull OpenSearchSwsResponse response) {
 
         var offset = getQueryOffset();
         var url = gatewayUri.toString().split("\\?")[0];
+        var requestParameter = toGateWayRequestParameter();
+        var id = createUriOffsetRef(url, requestParameter, offset);
+
+        var hasMoreResults = offset < response.getTotalSize();
+        var nextResults
+            = hasMoreResults
+                  ? createUriOffsetRef(url, requestParameter, offset + getQueryPageSize())
+                  : null;
+        var nextResultsBySortKey
+            = hasMoreResults
+                  ? getNextResultsBySortKey(response, requestParameter, url)
+                  : null;
+
+        var hasPreviousResults = offset > 0;
+        var previousResults
+            = hasPreviousResults
+                  ? createUriOffsetRef(url, requestParameter, offset - getQueryPageSize())
+                  : null;
```

Comment:
I would extract this variables in methods and pass method directly to builder, but its up to you.
For example :
`.withNextResults(calculateNextResult(someInput))`